x-pack/plugins/ml/public/application/contexts/ml/ml_server_info_context.tsx

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { type FC, type PropsWithChildren, createContext, useContext } from 'react';
+import type { NLPSettings } from '../../../../common/constants/app';
+
+export interface MlServerInfoContextValue {
+  // TODO add ML server info
+  nlpSettings: NLPSettings;
+}
+
+export const MlServerInfoContext = createContext<MlServerInfoContextValue | undefined>(undefined);
+
+export const MlServerContextProvider: FC<PropsWithChildren<MlServerInfoContextValue>> = ({
```

Comment:
I think `MlServerInfoContextProvider` would make it clearer what this contains and be inline with the names of the other items in this file.

Reply:
Renamed in 31ac25304966243ed6e7e317a72f7e0729b4809e